### DISCUSSION TITLE
RDP

### DISCUSSION THREAD

**Discussion by ruliusuifeng** [rustdesk/rustdesk#6316] [General]
### Bug Description

Using Windows to remote a Windows system virtual machine, RDP does not work, the connected IP shows localhost, and the black screen waits for five seconds and prompts "Your Remote Desktop Service session has ended"

### How to Reproduce

click RDP

### Expected Behavior

Correct remote, using the rdp that comes with windows can work

### Operating system(s) on local side and remote side

Windows 10 -> Windows 10 22H2

### RustDesk Version(s) on local side and remote side

1.2.4 -> 1.2.4

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/17266808/aaf7ea15-7295-4a63-a579-0ced38f7df13)
![image](https://github.com/rustdesk/rustdesk/assets/17266808/0006b3b3-aff7-496a-ad6f-4d6b2bb38ff4)

**Comment by 3gmgithub** [comment]
If you select your own local port using the TCP tunneling this works.
I had a similar problem because I was using local port 3389 for the local RDP.
I tried manually setting up a tcp tunnel with 3388 local and using 3389 remote, and the RDP worked fine then.
Using the RDP button does not work when you are using local 3389 for something else.  So it may make sense to update this to use random local port that is tested for current use before allocated.

**Reply by q1055145251** [reply]
I still encounter the aforementioned issue even when using a random port, unable to connect.  I can successfully connect to the remote computer's RDP on port 3389 using Zerotier, but it doesn't work when using RustDesk's RDP protocol to connect to the remote computer.

**Comment by turetafr** [comment]
I am having this problem as well. Compared to RDP over a reverse SSH tunnel, or RDP through a Wireguard tunnel, Rustdesk is sluggish.  The RDP tunneling just does not work and comes back with 0x904 every time.  Connection is fast, so I'm not sure why.  Using 1.3.6 and self-hosted on arch linux with nothing else on it.
